### PR TITLE
[CI:DOCS] Update podman-remote docs

### PR DESCRIPTION
**Pull request by Luap99**
Add support for multi level subcommands.
e.g. podman system connection.

Update the flags and add note for containers.conf.

### CODE REVIEW

**Comment by Luap99** [comment]
@baude @jwhonce PTAL

File: docs/remote-docs.sh [hunk 1]
```diff
@@ -80,12 +96,12 @@ function pub_pages() {
...
 
-    for c in "container" "image" "pod" "volume" ""; do
-        local cmd=${c:+-$c}
-        for s in $($PODMAN $c --help | sed -n '/^Available Commands:/,/^Flags:/p' | sed -e '1d;$d' -e '/^$/d' | awk '{print $1}'); do
-            $publisher $(echo $source/podman$cmd-$s.*)
-        done
-    done
+    while IFS= read -r cmd; do
+        file="podman-${cmd// /-}"
+        if [ -f $source/$file.* ]; then
```
Comment (edsantiago) [inline docs/remote-docs.sh:101]: Could you make this `-f $source/$file.1.md` instead? Also in the next line? `*` barfs horribly if there are `*~` or `*.bkp` or other such files.

File: docs/source/markdown/podman-remote.1.md [hunk 2]
```diff
@@ -17,7 +17,9 @@ Podman uses Buildah(1) internally to create container images. Both tools share i
 (not container) storage, hence each can use or manipulate images (but not containers)
 created by the other.
 
-Podman-remote provides a local client interacting with a Podman backend node through a RESTful API tunneled through a ssh connection. In this context, a Podman node is a Linux system with Podman installed on it and the API service activated. Credentials for this session can be passed in using flags, environment variables, or in `podman-remote.conf`
+Podman-remote provides a local client interacting with a Podman backend node through a RESTful API tunneled through a ssh connection. In this context, a Podman node is a Linux system with Podman installed on it and the API service activated. Credentials for this session can be passed in using flags, environment variables, or in `containers.conf`.
+
+The `containers.conf` has to be placed under `$HOME/.config/containers/containers.conf` on Linux and Mac and `%APPDATA%\containers\containers.conf` on Windows.
```
Comment (edsantiago) [inline docs/source/markdown/podman-remote.1.md:22]: Should probably be 'The containers.conf **file**'

**Comment by rhatdan** [comment]
LGTM

File: docs/source/markdown/podman-remote.1.md [hunk 3]
```diff
@@ -31,29 +33,19 @@ Remote connection name
...
+Podman prompts for the login password on the remote server.
 
-Alternate path for configuration file
-
-**--remote-host**=*ip*
-
-Remote host IP
-
-**--syslog**
+**--log-level**=*level*
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-remote.1.md:42]: I thought this was a global option and only documented in podman(1)?

Comment (Luap99) [reply docs/source/markdown/podman-remote.1.md:42]: podman(1) isn't included in the remote package. podman-remote(1) will be renamed to podman(1) while generating docs for remote.

**Comment by Luap99** [comment]
@edsantiago Thanks. 
```
$ LC_ALL=C diff -urq docs/build/{master,remote}
Only in docs/build/remote/darwin: podman-generate-kube.1
Only in docs/build/remote/darwin: podman-healthcheck-run.1
Only in docs/build/master/darwin: podman-help.1
Only in docs/build/remote/darwin: podman-manifest-add.1
Only in docs/build/remote/darwin: podman-manifest-create.1
Only in docs/build/remote/darwin: podman-manifest-inspect.1
Only in docs/build/remote/darwin: podman-manifest-push.1
Only in docs/build/remote/darwin: podman-manifest-remove.1
Only in docs/build/remote/darwin: podman-network-create.1
Only in docs/build/remote/darwin: podman-network-inspect.1
Only in docs/build/remote/darwin: podman-network-ls.1
Only in docs/build/remote/darwin: podman-network-rm.1
Only in docs/build/remote/darwin: podman-play-kube.1
Only in docs/build/remote/darwin: podman-system-connection-add.1
Only in docs/build/remote/darwin: podman-system-connection-default.1
Only in docs/build/remote/darwin: podman-system-connection-list.1
Only in docs/build/remote/darwin: podman-system-connection-remove.1
Only in docs/build/remote/darwin: podman-system-connection-rename.1
Only in docs/build/remote/darwin: podman-system-connection.1
Only in docs/build/remote/darwin: podman-system-df.1
Only in docs/build/remote/darwin: podman-system-info.1
Only in docs/build/remote/darwin: podman-system-prune.1
Files docs/build/master/darwin/podman.1 and docs/build/remote/darwin/podman.1 differ
Only in docs/build/remote/windows: podman-generate-kube.html
Only in docs/build/remote/windows: podman-healthcheck-run.html
Only in docs/build/master/windows: podman-help.html
Only in docs/build/remote/windows: podman-manifest-add.html
Only in docs/build/remote/windows: podman-manifest-create.html
Only in docs/build/remote/windows: podman-manifest-inspect.html
Only in docs/build/remote/windows: podman-manifest-push.html
Only in docs/build/remote/windows: podman-manifest-remove.html
Only in docs/build/remote/windows: podman-network-create.html
Only in docs/build/remote/windows: podman-network-inspect.html
Only in docs/build/remote/windows: podman-network-ls.html
Only in docs/build/remote/windows: podman-network-rm.html
Only in docs/build/remote/windows: podman-play-kube.html
Only in docs/build/remote/windows: podman-system-connection-add.html
Only in docs/build/remote/windows: podman-system-connection-default.html
Only in docs/build/remote/windows: podman-system-connection-list.html
Only in docs/build/remote/windows: podman-system-connection-remove.html
Only in docs/build/remote/windows: podman-system-connection-rename.html
Only in docs/build/remote/windows: podman-system-connection.html
Only in docs/build/remote/windows: podman-system-df.html
Only in docs/build/remote/windows: podman-system-info.html
Only in docs/build/remote/windows: podman-system-prune.html
Files docs/build/master/windows/podman.html and docs/build/remote/windows/podman.html differ
```

podman-help is missing but that was only a link to podman(1) anyway

**Comment by edsantiago** [comment]
You can add podman-help by fixing the 'grep' as per my suggested PR. Maybe we need it, maybe we don't, that is not my decision to make. I choose to believe that if a man page exists, it's for a reason.

File: docs/remote-docs.sh [hunk 4]
```diff
@@ -12,21 +12,24 @@ function usage() {
     echo >&2 "$0 PLATFORM TARGET SOURCES..."
     echo >&2 "PLATFORM: Is either linux, darwin or windows."
     echo >&2 "TARGET: Is the directory where files will be staged. eg, docs/build/remote/linux"
-    echo >&2 "SOURCES: Are the directories of source files. eg, docs/markdown"
+    echo >&2 "SOURCES: Are the directories of source files. eg, docs/source/markdown"
 }
 
 function fail() {
     echo >&2 -e "$@\n"
     usage
```
Comment (edsantiago) [inline docs/remote-docs.sh:20]: Please remove these two lines, they're not useful

File: docs/remote-docs.sh [hunk 5]
```diff
@@ -72,6 +75,22 @@ function html_fn() {
     pandoc --ascii --lua-filter=docs/links-to-html.lua -o $TARGET/${file%%.*}.html $markdown
 }
 
+# Run 'podman help' (possibly against a subcommand, e.g. 'podman help image')
+# and return a list of each first word under 'Available Commands', that is,
+# the command name but not its description.
+function podman_commands() {
+    $PODMAN help "$@" |\
+        awk '/^Available Commands:/{ok=1;next}/^Flags:/{ok=0}ok { print $1 }' |\
+        grep . | grep -v help
```
Comment (edsantiago) [inline docs/remote-docs.sh:84]: Please remove the `grep -v help`, so we keep the podman-help man page

**Comment by ashley-cui** [comment]
LGTM, thanks so much for fixing this up :)

File: docs/source/markdown/podman-remote.1.md [hunk 6]
```diff
@@ -17,7 +17,9 @@ Podman uses Buildah(1) internally to create container images. Both tools share i
 (not container) storage, hence each can use or manipulate images (but not containers)
 created by the other.
 
-Podman-remote provides a local client interacting with a Podman backend node through a RESTful API tunneled through a ssh connection. In this context, a Podman node is a Linux system with Podman installed on it and the API service activated. Credentials for this session can be passed in using flags, environment variables, or in `podman-remote.conf`
+Podman-remote provides a local client interacting with a Podman backend node through a RESTful API tunneled through a ssh connection. In this context, a Podman node is a Linux system with Podman installed on it and the API service activated. Credentials for this session can be passed in using flags, environment variables, or in `containers.conf`.
+
+The `containers.conf` file has to be placed under `$HOME/.config/containers/containers.conf` on Linux and Mac and `%APPDATA%\containers\containers.conf` on Windows.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-remote.1.md:22]: The below is the placement for rootless, not rootful no?
```suggestion
The `containers.conf` file should be placed under `$HOME/.config/containers/containers.conf` on Linux and Mac and `%APPDATA%\containers\containers.conf` on Windows.
```

File: docs/source/markdown/podman-remote.1.md [hunk 7]
```diff
@@ -31,29 +33,19 @@ Remote connection name
...
-
-Log messages above specified level: debug, info, warn, error (default), fatal or panic
-
-**--port**=*integer*
-
-Use an alternative port for the ssh connections.  The default port is 22
+**--identity**=*path*
 
-**--remote-config-path**=*path*
+Path to ssh identity file. If the identity file has been encrypted, podman prompts the user for the passphrase.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-remote.1.md:38]:
```suggestion
Path to ssh identity file. If the identity file has been encrypted, Podman prompts the user for the passphrase.
```

File: docs/source/markdown/podman-remote.1.md [hunk 8]
```diff
@@ -31,29 +33,19 @@ Remote connection name
...
+If no identity file is provided and no user is given, podman defaults to the user running the podman command.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-remote.1.md:39]:
```suggestion
If no identity file is provided and no user is given, Podman defaults to the user running the podman command.
```
sorry if I missed this last time.

File: docs/source/markdown/podman-remote.1.md (hunk 6 above)
Comment (edsantiago) [reply docs/source/markdown/podman-remote.1.md:22]: For podman-remote, shouldn't it always be HOME? Who would possibly want to run podman-remote as root?

Comment (ashley-cui) [reply docs/source/markdown/podman-remote.1.md:22]: I think its same for rootless and rootful on mac/windows since its just the client

Comment (Luap99) [reply docs/source/markdown/podman-remote.1.md:22]: There is no special reason why I did this. Should I include the root paths?

Comment (edsantiago) [reply docs/source/markdown/podman-remote.1.md:22]: >Should I include the root paths?

Oh please no. It's meaningless for windows/mac, and for anyone running podman-remote as root on linux, they just deserve what they get.

Comment (ashley-cui) [reply docs/source/markdown/podman-remote.1.md:22]: You don't need the root paths, this is correct

Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-remote.1.md]: @edsantiago that's why I was suggesting the change to should, I wasn't advocating adding the root location too.  I'm OK with it as is.

**Comment by TomSweeneyRedHat** [comment]
LGTM
Thanks @Luap99 !

**Comment by edsantiago** [comment]
/lgtm
/hold

Thank you @Luap99 and thanks @ashley-cui , @TomSweeneyRedHat for testing and review

**Pull request merged**